### PR TITLE
feat(EMS-711-712): Your buyer - Company or organisation - First and Last Name

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* Added first and last name fields
* Added validation
* Changed all widths of input to be one-half to fit design
* Added tests

### CODE REVIEW

File: e2e-tests/cypress/e2e/journeys/insurance/your-buyer/company-or-organisation.spec.js [hunk 1]
```diff
@@ -102,6 +104,27 @@ context('Insurance - Your Buyer - Company or organisation page - As an exporter,
...
+    firstNameField.label().should('exist');
+    cy.checkText(firstNameField.label(), FIELDS.COMPANY_OR_ORGANISATION[FIRST_NAME].LABEL);
+
+    firstNameField.input().should('exist');
+
+    lastNameField.label().should('exist');
+    cy.checkText(lastNameField.label(), FIELDS.COMPANY_OR_ORGANISATION[LAST_NAME].LABEL);
+
+    lastNameField.input().should('exist');
+  });
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-buyer/company-or-organisation.spec.js:126]: Can the two fields have their own `it`?

File: e2e-tests/cypress/e2e/journeys/insurance/your-buyer/validation/company-or-organisation-validation-first-last-name.spec.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          const { field, numberOfExpectedErrors, errorIndex } = ERROR_ASSERTIONS;
+
+          cy.submitAndAssertFieldErrors(field, null, errorIndex, numberOfExpectedErrors, errorMessage);
+        });
+      });
+    });
+
+    describe(`when ${FIRST_NAME_FIELD_ID} is correctly entered`, () => {
+      it('should not display validation errors', () => {
+        cy.keyboardInput(companyOrOrganisationPage[FIRST_NAME_FIELD_ID].input(), 'test');
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-buyer/validation/company-or-organisation-validation-first-last-name.spec.js:75]: Could we use E2E fixtures for the submitted value? Same with the last name field

File: e2e-tests/cypress/e2e/pages/insurance/your-buyer/companyOrOrganisation.js [hunk 3]
```diff
@@ -38,6 +40,18 @@ const companyOrOrganisation = {
     input: () => cy.get(`[data-cy="${WEBSITE}-input"]`),
     errorMessage: () => cy.get(`[data-cy="${WEBSITE}-error-message"]`),
   },
+  [FIRST_NAME]: {
+    heading: () => cy.get(`[data-cy="${FIRST_NAME}-heading"]`),
+    hint: () => cy.get(`[data-cy="${FIRST_NAME}-hint"]`),
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/pages/insurance/your-buyer/companyOrOrganisation.js:45]: Looks like we don't need `heading` or `hint`?

File: src/ui/server/controllers/insurance/your-buyer/validation/rules/first-name.test.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  describe(`when ${FIELD_ID} field is provided`, () => {
+    it('should not return validation errors', () => {
+      mockBody[FIELD_ID] = 'test';
+
+      const result = firstNameRule(mockBody, mockErrors);
+
+      expect(result).toEqual(mockErrors);
+    });
+  });
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/validation/rules/first-name.test.ts:47]: This test can just have one `it`, this helper function handles both scenarios 😄  Same for `last-name`

```js
it('should return the result of emptyFieldValidation', () => {
```

**Review by ttbarnes — DISMISSED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**